### PR TITLE
avoid halide_malloc when tracing

### PR DESCRIPTION
**Pull request by abadams**
It's a very minor speed-up to text-based tracing of local laplacian (2%)

This function was already allocating a page on the stack for the binary
case, and then not even using it in the text branch. May as well use it
and skip the halide_malloc call inside the printer.

### CODE REVIEW

**Comment by zvookin** [comment]
LGTM.

File: src/runtime/tracing.cpp [hunk 1]
```diff
@@ -169,7 +170,7 @@ WEAK int32_t default_trace(void *user_context, const halide_trace_event *e) {
 
         {
             ScopedSpinLock lock(&halide_trace_file_lock);
-            halide_print(user_context, ss.str());
+            halide_print(user_context, (const char *)buffer);
```
Comment (steven-johnson) [inline src/runtime/tracing.cpp]: This should probably have a call to

    halide_msan_annotate_memory_is_initialized(user_context, buffer, strlen(buffer));

to avoid false positives in msan mode.

Comment (dsharletg) [inline src/runtime/tracing.cpp]: Why is that needed here specifically? Surely there are loads of similar snippets of code in the runtime, that don't have msan annotations, including the printer itself, which is used for everything that gets printed from the runtime (yet I don't remember seeing msan problems from that code).

Comment (dsharletg) [inline src/runtime/tracing.cpp]: I see now, your other PR adds msan annotations in exactly these places!
